rabbitmq/check.py

```diff
@@ -353,8 +367,25 @@ def get_connections_stat(self, instance, base_url, object_type, vhosts, custom_t
         Collect metrics on currently open connection per vhost.
         """
         instance_proxy = self.get_instance_proxy(instance, base_url)
-        data = self._get_data(urlparse.urljoin(base_url, object_type), auth=auth,
-                              ssl_verify=ssl_verify, proxies=instance_proxy)
+
+        grab_all_data = True
+
+        if self._limit_vhosts(instance):
+            grab_all_data = False
+            data = []
+            for vhost in vhosts:
+                url = "vhosts/{}/{}".format(urllib.quote_plus(vhost), object_type)
+                try:
+                    data += self._get_data(urlparse.urljoin(base_url, url), auth=auth,
+                                          ssl_verify=ssl_verify, proxies=instance_proxy)
+                except Exception as e:
+                    # This will happen if there is no connection data to grab
+                    self.log.debug("Couldn't grab connection data from vhost, {}: {}".format(vhost, e))
+
+        # sometimes it seems to need to fall back to this
+        if grab_all_data or not len(data):
```

Comment:
I'd get rid of the `if self._limit_vhosts(instance):` block above and go through the vhosts - if vhosts is empty or there was a problem collecting data and `data` is empty, we can fall back to the global query. 
This would simply become:
```python
# even when vhosts are listed in the config file, sometimes it seems to need to fall back to this
if not data:
```

Reply:
vhosts will never be empty